crates/trie/src/state.rs

```diff
@@ -132,6 +132,40 @@ impl HashedPostState {
         Ok(this.sorted())
     }
 
+    /// Extend this hashed post state with contents of another.
+    /// Entries in the second hashed post state take precedence.
+    pub fn extend(&mut self, other: Self) {
+        // Merge accounts and insert them into extended state.
+        let mut accounts: HashMap<B256, Option<Account>> = HashMap::from_iter(
+            self.accounts
+                .drain(..)
+                .map(|(hashed_address, account)| (hashed_address, Some(account)))
+                .chain(
+                    self.destroyed_accounts.drain().map(|hashed_address| (hashed_address, None)),
+                ),
+        );
+        for (hashed_address, account) in other.accounts {
+            accounts.insert(hashed_address, Some(account));
+        }
+        for hashed_address in other.destroyed_accounts {
+            accounts.insert(hashed_address, None);
+        }
+        for (hashed_address, account) in accounts {
+            self.insert_account(hashed_address, account);
+        }
```

Comment:
do we have to drain, merge / overwrite, then re-insert here? can we overwrite in-place?

Reply:
for now - yes. i will make a follow-up PR to change how hashed state is represented internally

Reply:
ok, makes sense for now then